src/main/java/com/fcgl/madrid/search/controller/UserSearchController.java

```diff
@@ -30,9 +26,13 @@ public void setServices(UserSearchService categoryService) {
      *
      * TODO: Based on what we find out on elastic search. We might have to make an api request to elastic search from here as well.
      */
-    @GetMapping(path = "/query")
-    public ResponseEntity<Response> addUserSearchQuery(@Valid UserSearchRequest request) {
-        //Call a function is userSearchService
-        return null;
+    @PostMapping(path = "/query")
+    public ResponseEntity<Response<String>> addUserSearchQuery(@Valid @RequestBody UserSearchRequest request) {
+        return userSearchService.addQueryToTable(request);
+    }
+
+    @GetMapping(path = "/query") //TODO returns a 500 not sure why
+    public ResponseEntity<Response<String>> getQueriesForUser(@Valid @PathVariable Long userId) {
```

Comment:
Lets make an Object for this as well to keep with the pattern

class UserId {
...
}

Reply:
Done